app/stac_api/validators.py

```diff
@@ -187,3 +187,29 @@ def validate_item_properties_datetimes(
             properties_start_datetime,
             properties_end_datetime,
         )
+
+
+def validate_JSON_payload_write_requests(serializer_class, request_data):  # pylint: disable=invalid-name
+    '''
+    Validate JSON payload of write requests.
+    Args:
+         serializer_class: serializer class for validating and deserializing input and
+         for serializing output.
+
+         request_data: data of the request to be validated
+    Raises:
+        ValidateionError: Either in case of unexpected payload or if the payload contains
+        values for read-only fields.
+    '''
+    expected_payload = list(serializer_class().fields.keys())
+    expected_payload_read_only = list()
+
+    for field in serializer_class().fields.keys():
+        if serializer_class().fields[field].read_only:
+            expected_payload_read_only.append(field)
```

Comment:
This could be done in list comprehension `[field for field in serializer.fields if field.read_only]`

Reply:
:+1:

Reply:
Done